src/utp_stream.cpp

```diff
@@ -1795,10 +1791,11 @@ bool utp_socket_impl::send_pkt(int const flags)
 	// for the cwnd condition is to make sure the probe is surrounded by non-
 	// probes, to be able to distinguish a loss of the probe vs. just loss in
 	// general.
-	bool const mtu_probe = (m_mtu_seq == 0
-		&& m_write_buffer_size >= m_mtu_floor * 3
+	bool const mtu_probe = m_mtu_floor < m_mtu_ceiling
```

Comment:
when `m_mtu_floor` == `m_mtu_ceiling`, we're done, and we don't need to issue any more probes